cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -666,6 +670,13 @@ public LoadBalancer getLoadBalancer(String resourceGroupName, String loadBalance
         return handleAuthException(() -> azure.loadBalancers().getByResourceGroup(resourceGroupName, loadBalancerName));
     }
 
+    /**
+     * Returns the IP addresses associated with a particular Load Balancer in a particular Azure Resource Group.
+     *
+     * @param resourceGroupName the name of the resource group containing the load balancer
+     * @param loadBalancerName the name of the load balancer
+     * @return IP addresses
+     */
     public List<String> getLoadBalancerIps(String resourceGroupName, String loadBalancerName) {
         List<String> ipList = new ArrayList<>();
         List<String> publicIpAddressIds = getLoadBalancer(resourceGroupName, loadBalancerName).publicIPAddressIds();
```

Comment:
Ah, every call to `getPublicIpAddressById` results in an Azure api request being made. Since this is a for loop, the requests run in series, and will have to make a round trip for each one.

I've reworked it to compare the `resourceGroup` public IP objects with the Load balancer IP objects and only add those IP addresses:
```java
        List<String> idsAssociatedWithLoadBalancerPublicIps = getLoadBalancer(resourceGroupName, loadBalancerName).publicIPAddressIds();
        List<PublicIPAddress> publicIpAddressesInResourceGroup = getPublicIpAddresses(resourceGroupName);

        List<String> publicIps = new ArrayList<>();
        for (PublicIPAddress ipAddress : publicIpAddressesInResourceGroup) {
            if (idsAssociatedWithLoadBalancerPublicIps.contains(ipAddress.id())) {
                publicIps.add(ipAddress.ipAddress());
            }
        }

        return publicIps;
```

Which should be just two API calls, regardless of how many public IPs are added to the LB.

This does imply iterating through every Public IP in the Resource group, which will probably grow much faster than the number of publicIp addresses associated with a single LB, but we don't have to pay the network penalty for each IP in the resource group.

Reply:
Looks fine for me, is it tested? If yes, could you pls remove 
`// todo: test this`

Reply:
Discussed this a bit [with Steve below](https://github.com/hortonworks/cloudbreak/pull/9953#discussion_r609825153).

I'm going to remove the test comment and move this logic out of the `AzureClient` and in to the `AzureMetaDataCollector` in a follow up PR.